### DISCUSSION TITLE
How the props `class`, `className`, `classList` works on solidjs?

### DISCUSSION THREAD

**Discussion by PeterlitsZo** [solidjs/solid#948] [Q&A]
What will happen if a JSX element have three attrs `class`, `className` and `classList`?

I cannot find in source code. Could you please tell me where is the code, which deal with the three attrs?

Thank you very much!

---

The reason why I want to know: I want to have my own anchor element, which has its own styles, additional content. But I also want to make it merge those attrs `class`, `className` and `classList`. To let other basic components can change its style if it want.

**Comment by Abiti-936** [comment]
As I have tried class & className will become in one class & classList will Stay Unchanged because classList adds & removes classes based on conditions.

Just See: https://playground.solidjs.com/?hash=1670781606&version=1.3.13

**Reply by PeterlitsZo** [reply]
Great! It works well! But do you know where is the source :eyes: I try to find but I cannot find it...

**Reply by Abiti-936** [reply]
classList can be found in:
https://www.solidjs.com/docs/latest/api#classlist
https://www.solidjs.com/tutorial/bindings_classlist

Also Take a look at YouTube videos like:
https://m.youtube.com/watch?v=ZZ-a7B761Ds
https://m.youtube.com/watch?v=Xw9XMNn2k0o

**Reply by PeterlitsZo** [reply]
Sorry... I mean, source code... I know where is the document `>_<`. I try my best. I try to search, try to find in files... But I cannot find where is the source code.

**Comment by mdynnl** [comment]
`class` and `className` compile to the same thing while also merging both.

`classList` compiles to an effect that uses `ele.classList.toggle` or internal `classList` function that loops over the object value passed to it.

If both `class/className` and `classList` are used and if the former is dynamic, depending on the order of execution, the former can wipe out the result of the latter because it assigns to `className` prop of the element.

The compiled output can be seen in the playground output tab. Most exports in `solid-js/web` comes from https://github.com/ryansolid/dom-expressions. If you're interested in internals and how the compiler and rendering (sans reactivity) works, I suggest looking tests folder of the repo.

Hope that helps.

**Reply by mdynnl** [reply]
You can also head over to our discord https://discord.com/invite/solidjs for questions if that's your thing.
